transport/http/err_test.go

```diff
@@ -0,0 +1,64 @@
+package http_test
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/net/context"
+
+	transporthttp "github.com/go-kit/kit/transport/http"
+
+	"testing"
+)
+
+func TestClientEndpointEncodeError(t *testing.T) {
+	var (
+		sampleErr = errors.New("Oh no, an error")
+		enc       = func(r *http.Request, request interface{}) error {
+			return sampleErr
+		}
+		dec = func(r *http.Response) (response interface{}, err error) {
+			return nil, nil
+		}
+	)
+
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "localhost",
+		Path:   "/does/not/matter",
+	}
+
+	c := transporthttp.NewClient(
+		"GET",
+		u,
+		enc,
+		dec,
+	)
+
+	_, err := c.Endpoint()(context.Background(), nil)
+	if err == nil {
+		t.Log("err == nil")
+		t.Fail()
```

Comment:
I disagree, only slightly.  If it were t.Log/t.FailNow, then yes, it would be better as t.Fatal.  However, t.Fatal stops execution, whereas t.Fail allows it to continue.